protocols/gossipsub/src/types.rs

```diff
@@ -346,24 +340,90 @@ impl RpcOut {
     pub fn into_protobuf(self) -> proto::RPC {
         self.into()
     }
+
+    /// Returns true if the `RpcOut` is high priority.
+    pub(crate) fn high_priority(&self) -> bool {
+        matches!(
+            self,
+            RpcOut::Subscribe(_)
+                | RpcOut::Unsubscribe(_)
+                | RpcOut::Graft(_)
+                | RpcOut::Prune(_)
+                | RpcOut::IDontWant(_)
+        )
+    }
```

Comment:
It looks like we no longer need this method or the implementations for `Ord` and `PartialOrd`. IIUC, these were from an older version of #570, where BinaryHeap was used in [the Shared struct](https://github.com/sigp/rust-libp2p/blob/c3c7deab0663d282bbdf7a7df5c4ddecc58cb6aa/protocols/gossipsub/src/queue.rs#L43-L46).